### PR TITLE
Create scaling.php

### PR DESCRIPTION
**Pull request by iadj**
Guide on setting up a Yii2 application stack for scaling.

### CODE REVIEW

**Comment by samdark** [comment]
TODO for myself: link to this cookbook article from official Yii guide.

File: book/scaling.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Making Your Yii2 App Scalable
```
Comment (samdark) [inline book/scaling.md]: Sounds too general. "Scalable" may include something like database sharding as well. Do you have any idea for a more narrow name?

Comment (samdark) [inline book/scaling.md]: On the other hand, it explicitly says "application". Did you mean "PHP code only"?

File: book/scaling.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Also configure your application to use your hosted database server (hosted by i.e. Google SQL).
+
+## Configuring the Stack
+
+Set up a temporary single instance server to configure your application with.
+
+The application must be deployed to the server by Git, so that multiple servers will stay up to date with the application. This guide recommends the following process:
+
+* `git clone` the application into the configured 'www' directory
```
Comment (samdark) [inline book/scaling.md]: In this case you need to secure `.git` directory not to be accessible via web. Else you're exposing app sources.

Comment (samdark) [inline book/scaling.md]: See https://www.smashingmagazine.com/2009/09/svn-strikes-back-a-serious-vulnerability-found/

File: book/scaling.md (hunk 1 above)
Comment (iadj) [inline book/scaling.md]: I think 'Configuring a Yii2 Application for Autoscaling' would be more precise.

File: book/scaling.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+* Set up a cron job to `git pull` the directory every minute.
+* Set up a cron job to `composer update` the directory every minute.
```
Comment (samdark) [inline book/scaling.md]: Overall it's not a good idea to update automatically. And even if it's done like that, it's better to use centralized deploy instance which updates all the servers and does it by incoming github/gitlab commit hook.

File: book/scaling.md (hunk 2 above)
Comment (iadj) [inline book/scaling.md]: Good find, that's an error on my part. The .git directory itself should be located somewhere secure, and the pulled project files should go in the public directory.

File: book/scaling.md (hunk 1 above)
Comment (samdark) [inline book/scaling.md]: Agree.

File: book/scaling.md (hunk 3 above)
Comment (iadj) [inline book/scaling.md]: Git hooks are definitely more efficient. If anyone knows the most efficient method please replace those steps!

**Comment by samdark** [comment]
Merged and added to index. Thank you!

**Comment by samdark** [comment]
https://github.com/samdark/yii2-cookbook/blob/master/book/scaling.md

**Comment by iadj** [comment]
Awesome!

**Comment by badams** [comment]
Not sure if this is the right place to post this..

👍 This sounds like a great topic for the cookbook, here are a few things that I think are worth considering:

* Not sure if deployment needs to be included in this recipe - I think the best solution for this is something like capistrano/deployer running from a CI/build server (which probably are another two separate recipes in themselves).
* Need a strategy for serving of assets (assuming they will be served from your application servers) a static path will be required, I would suggest the[ example in the docs](http://www.yiiframework.com/doc-2.0/yii-web-assetmanager.html#%24hashCallback-detail) as a solution.
* Handling Logging, probably want all the application servers logging to a single place, what solutions are you using here?

**Comment by samdark** [comment]
@badams great suggestions.

**Comment by iadj** [comment]
@badams Awesome suggestions. Would really make it complete.

**Comment by iadj** [comment]
@badams I added a Logging solution made by @JackyChan to the resources https://github.com/samdark/yii2-cookbook/pull/115

How would you suggest doing the serving of assets? The link you included isn't quite clear to me.

**Comment by badams** [comment]
@iadj Nice work :) take a look at my comments here https://github.com/samdark/yii2-cookbook/pull/117#issuecomment-225062579

**Pull request merged**